webapp/channels/src/components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -36,508 +63,301 @@ type Props = PropsFromRedux & {
-    }: DesktopNotificationProps) {
-        const currentUserNotifyProps = this.props.currentUser.notify_props;
-
-        if (
-            desktopNotifyLevel === getDefaultDesktopNotificationLevel(currentUserNotifyProps, this.isGM()) &&
-            desktopNotifySound === getDefaultDesktopNotificationSound(currentUserNotifyProps) &&
-            desktopSound === getDefaultDesktopSound(currentUserNotifyProps) &&
-            desktopThreadsNotifyLevel === getDefaultDesktopThreadsNotifyLevel(currentUserNotifyProps)
-        ) {
-            return true;
-        }
-        return false;
-    }
-
-    isGM() {
-        return this.props.channel.type === 'G';
-    }
-
-    verifyPushNotificationsSettingSameAsGlobal({
-        pushNotifyLevel,
-        pushThreadsNotifyLevel,
-    }: PushNotificationProps) {
-        const currentUserNotifyProps = this.props.currentUser.notify_props;
-
-        if (
-            pushNotifyLevel === getDefaultPushNotifyLevel(currentUserNotifyProps, this.isGM()) &&
-            pushThreadsNotifyLevel === getDefaultPushThreadsNotifyLevel(currentUserNotifyProps, this.isGM())
-        ) {
-            return true;
+        if (settingName === 'push') {
+            setSettings({...settings, push: defaultSettings.desktop, push_threads: defaultSettings.push_threads || settings.push_threads});
         }
-        return false;
-    }
-
-    getStateFromNotifyProps(currentUserNotifyProps: UserNotifyProps, channelMemberNotifyProps?: ChannelMemberNotifyProps) {
-        let ignoreChannelMentionsDefault: ChannelNotifyProps['ignore_channel_mentions'] = IgnoreChannelMentions.OFF;
-
-        const desktopNotifyLevelDefault: ChannelNotifyProps['desktop'] = getDefaultDesktopNotificationLevel(currentUserNotifyProps, this.isGM());
-        const pushNotifyLevelDefault: ChannelMemberNotifyProps['push'] = getDefaultPushNotifyLevel(currentUserNotifyProps, this.isGM());
-        const pushThreadsNotifyLevelDefault: ChannelMemberNotifyProps['push_threads'] = getDefaultPushThreadsNotifyLevel(currentUserNotifyProps, this.isGM());
-
-        const channelDesktopNotifyProps = channelMemberNotifyProps?.desktop || NotificationLevels.DEFAULT;
-        let desktopNotifyLevel = desktopNotifyLevelDefault;
-        if (channelDesktopNotifyProps !== NotificationLevels.DEFAULT) {
-            desktopNotifyLevel = channelDesktopNotifyProps;
-        }
-
-        const channelPushNotifyProps = channelMemberNotifyProps?.push || NotificationLevels.DEFAULT;
-        let pushNotifyLevel = pushNotifyLevelDefault;
-        if (channelPushNotifyProps !== 'default') {
-            pushNotifyLevel = channelPushNotifyProps;
+    }, [settings, props.currentUser]);
+
+    const handleChange = useCallback((values: Record<string, string>) => {
+        setSettings({...settings, ...values});
+    }, [settings]);
+
+    const handleMobileSettingsChange = useCallback(() => {
+        setMobileSettingsSameAsDesktop(!mobileSettingsSameAsDesktop);
```

Comment:
My comment on `setSettings` also applies to `setMobileSettingsSameAsDesktop` as well